src/main/java/io/orkes/conductor/client/automator/TaskRunner.java

```diff
@@ -67,12 +73,14 @@ class TaskRunner {
             String workerNamePrefix,
             int threadCount,
             int taskPollTimeout) {
+        this.worker = worker;
         this.eurekaClient = eurekaClient;
         this.taskClient = taskClient;
         this.updateRetryCount = updateRetryCount;
         this.taskToDomain = taskToDomain;
         this.threadCount = threadCount;
         this.taskPollTimeout = taskPollTimeout;
+        this.permits = new Semaphore(threadCount);
```

Comment:
Given that `TaskRunner` is already within a `ThreadPollExecutor`, I think this Semaphore is redundant. Let me try to make a PR into this branch with an alternative for it.